### PR TITLE
Dimensions 📐 - updated Docs 📑 & new snack examples ✨

### PR DESCRIPTION
**Pull request by DaniAkash**
Belongs to https://github.com/facebook/react-native-website/issues/1579

Inspiration - To address this [stackoverflow question](https://stackoverflow.com/questions/44978804/whats-the-difference-between-window-and-screen-in-the-dimensions-api)

Contains the snack with both class & functional component examples and also a bit of update to the documentation regarding screen, window & event listener.

Pending items in this PR:
- [ ] `useDimensions` hook
- [ ] Info regarding scaling
- [ ] Verify if rounding is necessary for using Dimensions

### CODE REVIEW

File: docs/dimensions.md [hunk 1]
```diff
@@ -2,16 +2,129 @@
...
+    alignItems: "center"
+  }
+});
 ```
+
+<block class="endBlock syntax" />
+
+### React Native Hooks
+
+You can also try [useDimensions](https://github.com/react-native-community/react-native-hooks#usedimensions) hook from [React native hooks](https://github.com/react-native-community/react-native-hooks) library which makes handling screen/window size changes much simpler.
```
Comment (jeremy-deutsch) [inline docs/dimensions.md]: RN actually ships its own `useWindowDimensions` hook now:
https://github.com/facebook/react-native/blob/8d57691a606ce81e6d861f93ccc11875088f58b6/Libraries/Utilities/useWindowDimensions.js

The docs should probably describe that hook here instead

Comment (DaniAkash) [reply docs/dimensions.md]: Thanks, I didn't knew!

File: docs/dimensions.md [hunk 2]
```diff
@@ -42,9 +157,9 @@ Example: `var {height, width} = Dimensions.get('window');`
 
 **Parameters:**
 
-| Name      | Type     | Required | Description                                                                                   |
-| ------    | ------   | -------- | ----------------------------------------------------------------------------------------------|
-| dim       | string   | Yes      | Name of dimension as defined when calling `set`. @returns {Object?} Value for the dimension.  | 
+| Name | Type   | Required | Description                                                                                  |
+| ---- | ------ | -------- | -------------------------------------------------------------------------------------------- |
+| dim  | string | Yes      | Name of dimension as defined when calling `set`. @returns {Object?} Value for the dimension. |
```
Comment (jeremy-deutsch) [inline docs/dimensions.md]:
```suggestion
| dim  | string | Yes      | Name of dimension as defined when calling `set`. |

Returns an object with four number fields: `width`, `height`, `scale`, and `fontScale`.
```

`scale` and `numberScale` should maybe be documented better here, but honestly, I don't really know much about them!

Comment (DaniAkash) [reply docs/dimensions.md]: Indeed, these areas could use some more detail. I'll try them out and update the docs

**Comment by Simek** [comment]
In the first example documentation is suggesting rounding `Dimensions` values before usage. This part was added in #1274. I don't know the exact reason behind this but I would assume that in some cases/on some devices this value could return float. 

Currently proposed examples do not utilize `Math.round`. I'm not sure which approach to handling those values is better but I think that someone should decide about that for the consistency and after that some parts of code should be updated.

**Comment by DaniAkash** [comment]
Closing it in favour of https://github.com/facebook/react-native-website/pull/1671

**Pull request closed without merging**